osu.Game/Tests/Visual/OnlinePlay/TestRoomRequestsHandler.cs

```diff
@@ -277,11 +278,18 @@ private Room cloneRoom(Room source)
             var result = JsonConvert.DeserializeObject<Room>(JsonConvert.SerializeObject(source));
             Debug.Assert(result != null);
 
-            // Playlist item IDs aren't serialised.
+            // Playlist item IDs and beatmaps aren't serialised.
             if (source.CurrentPlaylistItem.Value != null)
+            {
+                result.CurrentPlaylistItem.Value = result.CurrentPlaylistItem.Value.With(new Optional<IBeatmapInfo>(source.CurrentPlaylistItem.Value.Beatmap));
```

Comment:
Commentary for other readers: normally you'd expect such code to skip the explicit `Optional<T>` construction because `Optional<T>` defines an implicit conversion from `T`, but in this instance that implicit conversion cannot be applied because here `T` is `IBeatmapInfo` and the language spec forbids such cases:

> For a given source type $S$ and target type $T$, if $S$ or $T$ are nullable value types, let $S_0$ and $T_0$ refer to their underlying types; otherwise, $S_0$ and $T_0$ are equal to $S$ and $T$ respectively. A class or struct is permitted to declare a conversion from a source type $S$ to a target type $T$ only if all of the following are true:
> - $S_0$ and $T_0$ are different types.
> - Either $S_0$ or $T_0$ is the instance type of the class or struct that contains the operator declaration.
> - **Neither $S_0$ nor $T_0$ is an *interface\_type*.**
> - Excluding user-defined conversions, a conversion does not exist from $S$ to $T$ or from $T$ to $S$.

(source: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/conversions#1052-permitted-user-defined-conversions)